lib/rules/compat-engines.ts

```diff
@@ -248,11 +248,14 @@ export default createRule("compat-engines", {
     ) {
       const currModules = [...modules, `${name}@${ver}`];
 
-      processMeta(ctx, getMetaFromNodeModules(name, ver, { context }));
-
-      if (!ctx.hasInvalid() && ctx.isAllProcessed()) {
-        return;
+      const nodeModulesMeta = getMetaFromNodeModules(name, ver, { context });
```

Comment:
The `node_modules` installed package could not using the satisfied version, `"semver": "7.3.5"` for example in test case https://github.com/ota-meshi/eslint-plugin-node-dependencies/blob/328fb26baca88ac5372c37055206de8d5a1948d2/tests/lib/rules/compat-engines.ts#L211.